subsys/mgmt/ec_host_cmd/backends/ec_host_cmd_backend_espi.c

```diff
@@ -0,0 +1,135 @@
+/*
+ * Copyright (c) 2022 Google LLC
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <string.h>
+
+#include <zephyr/device.h>
+#include <zephyr/drivers/espi.h>
+#include <zephyr/logging/log.h>
+#include <zephyr/mgmt/ec_host_cmd/ec_host_cmd.h>
+#include <zephyr/mgmt/ec_host_cmd/backend.h>
+
+LOG_MODULE_REGISTER(host_cmd_espi, CONFIG_EC_HC_LOG_LEVEL);
+
+#define RX_HEADER_SIZE (sizeof(struct ec_host_cmd_request_header))
+
+struct ec_host_cmd_espi {
+	/* eSPI device instance */
+	const struct device *espi_dev;
+	/* Context for read operation */
+	struct ec_host_cmd_rx_ctx *rx_ctx;
+	/* Transmit buffer */
+	struct ec_host_cmd_tx_buf *tx;
+	/* eSPI callback */
+	struct espi_callback espi_cb;
+};
+
+#define EC_HOST_CMD_ESPI_DEFINE(_name)                                                             \
+	static struct ec_host_cmd_espi _name##_hc_espi;                                            \
+	struct ec_host_cmd_backend _name = {                                                       \
+		.api = &ec_host_cmd_api,                                                           \
+		.ctx = (struct ec_host_cmd_espi *)&_name##_hc_espi,                                \
+	}
+
+static void espi_handler(const struct device *dev, struct espi_callback *cb,
+			 struct espi_event espi_evt)
+{
+	struct ec_host_cmd_espi *hc_espi = CONTAINER_OF(cb, struct ec_host_cmd_espi, espi_cb);
+	uint16_t event_type = (uint16_t)espi_evt.evt_details;
+	/* tx stores the shared memory buf pointer and size, so use it */
+	const struct ec_host_cmd_request_header *rx_header = hc_espi->tx->buf;
+	const size_t shared_size = hc_espi->tx->len_max;
```

Comment:
Regarding the comment of size for different backends. The eSPI can have a different shared memory size, if it's smaller than the "maximum" buffer size, it may overflow.

Reply:
I don't think so, the sizes are checked before memcpy.